src/server/gc_worker/compaction_filter.rs

```diff
@@ -915,21 +978,48 @@ pub mod tests {
         must_prewrite_put(&engine, b"zkey", &value, b"zkey", 100);
         must_commit(&engine, b"zkey", 100, 110);
         gc_runner.safe_point(50).gc(&raw_engine);
+        assert_eq!(
+            MVCC_VERSIONS_HISTOGRAM
+                .with_label_values(&[TXN_KEYMODE])
+                .get_sample_sum(),
+            0 as f64
+        );
+        assert_eq!(
+            GC_COMPACTION_FILTERED
+                .with_label_values(&[TXN_KEYMODE])
+                .get(),
+            0
+        );
+
         must_get(&engine, b"zkey", 110, &value);
 
         // GC can't delete keys before the safe ponit if they are latest versions.
         gc_runner.safe_point(200).gc(&raw_engine);
+
         must_get(&engine, b"zkey", 110, &value);
 
         must_prewrite_put(&engine, b"zkey", &value, b"zkey", 120);
         must_commit(&engine, b"zkey", 120, 130);
 
         // GC can't delete the latest version before the safe ponit.
         gc_runner.safe_point(115).gc(&raw_engine);
+
         must_get(&engine, b"zkey", 110, &value);
 
         // GC a version will also delete the key on default CF.
         gc_runner.safe_point(200).gc(&raw_engine);
+        assert_eq!(
+            MVCC_VERSIONS_HISTOGRAM
+                .with_label_values(&[TXN_KEYMODE])
+                .get_sample_sum(),
+            4_f64
```

Comment:
Seems that it is not expected to be `4`.

Reply:
```MVCC_VERSIONS_HISTOGRAM.with_label_values(&[TXN_KEYMODE]).observe(1)```
has been executed 4 times.
So ```MVCC_VERSIONS_HISTOGRAM.with_label_values(&[TXN_KEYMODE]).get_sample_sum()``` is ```4```